presto-hive/src/main/java/com/facebook/presto/hive/HiveWriterFactory.java

```diff
@@ -366,11 +369,10 @@ public HiveWriter createWriter(Page partitionColumns, int position, OptionalInt
         }
 
         if (sortingFileWriterFactory.isPresent()) {
-            checkState(bucketNumber.isPresent(), "missing bucket number for sorted table write");
             hiveFileWriter = sortingFileWriterFactory.get().createSortingFileWriter(
                     path,
                     hiveFileWriter,
-                    bucketNumber.getAsInt(),
+                    bucketNumber.orElse(random.nextInt()),
```

Comment:
what if two writers write to the same fileNumber?

Reply:
That is okay because this `fileNumber` is only used to determine which temp subdirectory we should write temp file into.

From another perspective, temp file name is generated by appending the generated prefix with uuid, so we would not write to the same file.

Reply:
To make it less confusing, maybe we should directly pass the subdirectory number instead of file number?

Reply:
It might be better to make a comment on this and make sure the number is between 0 to some small numbers (rather than 2^31).